java-squid/src/main/java/org/sonar/java/se/EqualConstraint.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se;
+
+import org.sonar.java.se.ConstraintManager.BooleanConstraint;
+
+import javax.annotation.Nullable;
+
+public class EqualConstraint extends AbstractBinaryConstraint {
+
+  protected EqualConstraint(SymbolicValue v1, SymbolicValue v2) {
+    super(v1, v2);
+  }
+
+  @Override
+  protected EqualConstraint inverted() {
```

Comment:
Sounds poorly named to me : AFAIU inverted would mean !equal whereas here it returns the symetric constraint ?